### PR TITLE
pass-gen-2

### PR DESCRIPTION
**Pull request by TahaBaber-devsinc**
password genertor

### CODE REVIEW

File: pass_gen.arb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# password gen
```
Comment (beinghaziq) [inline pass_gen.arb:3]: make it more descriptive

Comment (TahaBaber-devsinc) [reply pass_gen.arb:3]: Noted

File: pass_gen.arb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  def generator_char_arr(num, schar, _len)
+    # initialize an empty array
+    # array << [characters] if @num == y
+    # return array
+    char_arr = [*('a'..'z'), *('A'..'Z')]
+    char_arr << [*('0'..'9')] if num == 'y'
+    char_arr << [*('!'..'~')] - [*('0'..'9')] if schar == 'y'
+    char_arr.flatten
```
Comment (beinghaziq) [inline pass_gen.arb:31]: Make them as constants. e.g. ALPHABETS = `[*('a'..'z'), *('A'..'Z')]`

File: pass_gen.arb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @pass = String.new
+  end
+
+  def generator_input
+    puts "Do you want to add numbers?\nEnter y to agree"
+    num = gets.chomp
+    puts "Do you want to add special characters?\nEnter y to agree"
+    schar = gets.chomp
+    puts 'Enter length of Password: '
+    len = Integer(gets, exception: false)
```
Comment (beinghaziq) [inline pass_gen.arb:15]: loop through them until user enter a valid value.

**Review by beinghaziq — CHANGES_REQUESTED** [review]
Added some comments

**Review by beinghaziq — APPROVED** [review]

File: pass_gen.arb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  def correct_length
+    len = Integer
+    loop do
+      puts 'Enter length of Password: '
+      len = Integer(gets, exception: false)
+
+      if len.is_a? Integer
+
+        len > 0 && len < 26 ? break : puts('Invalid input. Length should be between 1-25')
```
Comment (beinghaziq) [inline pass_gen.arb:20]: create constants for 0 and 26

**Review by beinghaziq — APPROVED** [review]
Conditionally approving

**Pull request merged**